loopy/schedule/tools.py

```diff
@@ -65,6 +74,62 @@ def temporaries_written_in_subkernel(kernel, subkernel):
             for tv in kernel.id_to_insn[insn_id].write_dependency_names()
             if tv in kernel.temporary_variables)
 
+
+def args_read_in_subkernel(kernel, subkernel):
```

Comment:
Any way that you could merge the `args` and `temporaries` versions? They look awfully similar.